streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -1584,7 +1646,43 @@ public <V> void shouldHandleKeyQuery(
         );
 
         final V result1 = queryResult.getResult();
-        final Integer integer = valueExtactor.apply(result1);
+        final Integer integer = (Integer) result1;
+        assertThat(integer, is(expectedValue));
+        assertThat(queryResult.getExecutionInfo(), is(empty()));
+        assertThat(queryResult.getPosition(), is(POSITION_0));
+    }
+
+    public <V> void shouldHandleTimestampedKeyQuery(
+            final Integer key,
+            final Integer expectedValue) {
```

Comment:
I think we should pass in `ValueAndTimestamp` as expected result and also verify that we get the correct timestamp back. (Also rename `expectedValue` to `expectedValueAndTimestamp`.)

Reply:
But in the end we should compare the value in `ValueAndTimestamp`?

Reply:
We should compare both (what happens automatically if you compare `ValueAndTimestamp` type). Thus, you need to set the right timestamp on expected `ValueAndTimestamp`.

The ts comes from the data, and is set in `before()` to `WINDOW_START + Duration.ofMinutes(2).toMillis() * i` when input data is written.

Reply:
ok